frontend/pkg/frontend/frontend.go

```diff
@@ -431,6 +509,227 @@ func (f *Frontend) ArmResourceCreateOrUpdate(writer http.ResponseWriter, request
 	}
 }
 
+// clusterResourceIdFromNodePoolResourceID strips the nodepool part of the full resource ID
+// returning the full cluster resourceID.
+// given a full resource ID like
+// /subscriptions/00000000-0000-0000-0000-000000000000/resourcegroups/dev-test-rg/providers/microsoft.redhatopenshift/hcpopenshiftclusters/aldo-cluster/nodepools/aldo-nodepool
+// it returns the resourceID of the cluster
+// /subscriptions/00000000-0000-0000-0000-000000000000/resourcegroups/dev-test-rg/providers/microsoft.redhatopenshift/hcpopenshiftclusters/aldo-cluster
+func clusterResourceIdFromNodePoolResourceID(resourceID string) string {
+	i := strings.Index(resourceID, "/nodepools")
+	clusterResourceId := resourceID[0:i]
+	return clusterResourceId
+}
+
+func (f *Frontend) CreateNodePool(writer http.ResponseWriter, request *http.Request) {
+	var err error
```

Comment:
if we're assigning err near the top, why declare it as a var here?

Reply:
We're inconsistent in the functions in this regard.  Delete doesn't have it.

Reply:
Agree. Left over from a previous block copy.